fmatch/matcher.py

```diff
@@ -230,8 +230,7 @@ def get_agg_metric_query(
         metric_queries = []
         not_queries = [
             ~Q("match", **{not_item_key: not_item_value})
-            for not_item in metrics.get("not", [])
-            for not_item_key, not_item_value in not_item.items()
+            for not_item_key, not_item_value in metrics.get("not", {}).items()
```

Comment:
But if we look at the new logic, it is changing the interpretation to a single dict with multiple key value pairs
```
{'labels.container.keyword': 'kube-rbac-proxy', 'labels.namespace.keyword': 'xyz'}
```
and the generate query would be same
```
[Bool(must_not=[Match(labels__container__keyword='kube-rbac-proxy')]), Bool(must_not=[Match(labels__namespace__keyword='xyz')])]
```

But the yaml structure for `not` would change to
```
      not:
        labels.container.keyword: kube-rbac-proxy
        labels.namespace.keyword: xyz
```